### PR TITLE
Bugfix: replication DLQ handler mem allocation

### PR DESCRIPTION
**Pull request by wxing1292**
**What changed?**
* Fix replication DLQ handler mem allocation
* Add more tests
* Optimize other use of make slice

**Why?**
Fix a small bug in replication DLQ handler mem allocation

**How did you test it?**
Existing tests

**Potential risks**
N/A

### CODE REVIEW

File: common/persistence/cassandra/cassandraHistoryPersistence.go [hunk 1]
```diff
@@ -463,12 +463,13 @@ func (h *cassandraHistoryV2Persistence) GetHistoryTree(
 	}
 	query := h.session.Query(v2templateReadAllBranches, treeID)
 
+	pageSize := 100
```
Comment (sergeybykov) [inline common/persistence/cassandra/cassandraHistoryPersistence.go:466]: Does this need to be configurable or passed here as an argument?

File: common/persistence/sql/sqlHistoryManager.go [hunk 2]
```diff
@@ -480,15 +480,14 @@ func (m *sqlHistoryV2Manager) GetHistoryTree(
...
-	branches := make([]*persistencespb.HistoryBranch, 0)
-
 	rows, err := m.db.SelectFromHistoryTree(ctx, sqlplugin.HistoryTreeSelectFilter{
 		TreeID:  treeID,
 		ShardID: *request.ShardID,
 	})
 	if err == sql.ErrNoRows || (err == nil && len(rows) == 0) {
 		return &p.GetHistoryTreeResponse{}, nil
 	}
+	branches := make([]*persistencespb.HistoryBranch, 0, len(rows))
```
Comment (sergeybykov) [inline common/persistence/sql/sqlHistoryManager.go:490]: Do we need a max capacity limit here instead of `len(rows)`?

**Review by sergeybykov — APPROVED** [review]

File: common/persistence/sql/sqlHistoryManager.go (hunk 2 above)
Comment (wxing1292) [reply common/persistence/sql/sqlHistoryManager.go:490]: `make([]*persistencespb.HistoryBranch, 0, len(rows))` this is allocating mem with max cap of `len(rows)`

the code down below `branches = append(branches, treeInfo.BranchInfo)` will reallocate mem if above limit is reached

File: common/persistence/cassandra/cassandraHistoryPersistence.go (hunk 1 above)
Comment (wxing1292) [reply common/persistence/cassandra/cassandraHistoryPersistence.go:466]: not according to the current use case

**Pull request merged**